src/dataframe/dataframe.jl

```diff
@@ -510,6 +510,28 @@ function Base.getindex(df::DataFrame, ::typeof(!), col_ind::SymbolOrString)
 end
 
 # df[MultiRowIndex, MultiColumnIndex] => DataFrame
+
+function _threaded_getindex(selected_rows::AbstractVector,
```

Comment:
this is the change that is leading to a regression.
Regression is minimal (microsecond level) and only if there are very many columns. Essentially - I have removed code duplication we had in the past differentiating the case when `:` is used for column selection (now `:` falls back to general column selector). This leads to a slight regression, but I thought that given it is very small it is OK to accept it.